core/scripts/IntegrationTests.js

```diff
@@ -0,0 +1,567 @@
+          } catch (error) {
+            continue;
+          }
+        }
+        try {
+          // the liquidator should always try withdraw, even if disputed
+          await expiringMultiParty.withdrawLiquidation(liquidation.id, liquidation.sponsor, { from: liquidator });
+        } catch (error) {
+          continue;
+        }
+      }
+    }
+
+    // STEP 9): expire the contract and settle positions
+    console.log("Advancing time and settling contract");
+    await expiringMultiParty.setCurrentTime(expirationTime.toNumber() + 1);
+    await mockOracle.setCurrentTime(expirationTime.toNumber() + 1);
+
+    await expiringMultiParty.expire();
+
+    // After expiration the oracle needs to settle the price. Push a price of 1 usd per collateral means that each
+    // token is redeemable for 1 unit of underlying.
+    const oracleTime = await expiringMultiParty.getCurrentTime();
+    await mockOracle.pushPrice(constructorParams.priceFeedIdentifier, oracleTime.subn(1).toString(), settlementPrice);
+
+    // STEP 10): all users withdraw their funds
+    // Settle the liquidator
+    await expiringMultiParty.settleExpired({ from: liquidator });
+    assert.equal(await syntheticToken.balanceOf(liquidator), "0");
+
+    // Settle the disputer
+    await expiringMultiParty.settleExpired({ from: disputer });
+    assert.equal(await syntheticToken.balanceOf(disputer), "0");
+
+    // Settle token holders
+    for (const tokenHolder of tokenHolders) {
+      await expiringMultiParty.settleExpired({ from: tokenHolder });
+      assert.equal(await syntheticToken.balanceOf(tokenHolder), "0");
+    }
+
+    // Settle token sponsors
+    for (const sponsor of sponsors) {
+      await expiringMultiParty.settleExpired({ from: sponsor });
+      assert.equal(await syntheticToken.balanceOf(sponsor), "0");
+    }
+
+    console.log("All accounts have been able to withdraw without revert!");
+    console.table({
+      iterations: numIterations,
+      positionsCreated: positionsCreated,
+      tokensTransferred: tokenTransfers,
+      additionalDepositsMade: depositsMade,
+      redemptionsMade: redemptionsMade,
+      liquidations: liquidationsObject.length,
+      disputedLiquidations: liquidationsObject.filter(liquidation => liquidation.disputed).length,
+      finalBalanceDrift: (await collateralToken.balanceOf(expiringMultiParty.address)).toNumber()
+    });
+
+    // STEP 11): ensure all funds were taken from the contract.
+    // The main assertion we can check is that all users were able to call `settleExpired` without the contract
+    // locking up. Additionally, if all book keeping has gone correctly, there should be no collateral left in
+    // the expiring multi party as this has all be withdrawn by token holders.
+    assert.equal((await collateralToken.balanceOf(expiringMultiParty.address)).toString(), "0");
+  });
+  it("Iterative full life cycle test with unfriendly numbers", async function() {
```

Comment:
Can we separate this second test into a follow up PR to keep this one shorter?

Reply:
done